pkg/frontend/mysql_cmd_executor.go

```diff
@@ -2096,6 +2111,81 @@ func (cwft *TxnComputationWrapper) Compile(requestCtx context.Context, u interfa
 	if err != nil {
 		return nil, err
 	}
+	// check if it is necessary to initialize the temporary engine
+	if cwft.compile.NeedInitTempEngine(cwft.ses.InitTempEngine) {
```

Comment:
add Get method for  `InitTempEngine` protected by mutex to remove the potential data race.

Reply:
done